test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment.pass.cpp

```diff
@@ -16,98 +16,120 @@
+        //T vals[n] = { n random numbers between 0 and 4 };
 
+        ::std::srand(42);
         Size segment_length = 1;
-        for (Size i = 0; i != n; )
+        Size j = 0;
+        for (Size i = 0; i != n; ++i)
         {
-          for (Size j = 0; j != 4*segment_length && i != n; ++j)
-          {
-              host_keys[i] = j/segment_length + 1;
-              host_vals[i] = 1;
-              host_val_res[i] = 0;
-              ++i;
-          }
-          ++segment_length;
+            host_keys[i] = j / segment_length + 1;
+            host_vals[i] = rand() % 5;
+            host_val_res[i] = 0;
+            ++j;
+            if (j == 4 * segment_length)
+            {
+                ++segment_length;
+                j = 0;
+            }
         }
     }
 
 #ifdef DUMP_CHECK_RESULTS
     template <typename Iterator, typename Size>
     void display_param(const char* msg, Iterator it, Size n)
     {
-        std::cout << msg;
+        ::std::cout << msg;
         for (Size i = 0; i < n; ++i)
         {
             if (i > 0)
-                std::cout << ", ";
-            std::cout << it[i];
+                ::std::cout << ", ";
+            ::std::cout << it[i];
         }
-        std::cout << std::endl;
+        ::std::cout << ::std::endl;
     }
 #endif // DUMP_CHECK_RESULTS
 
     template <typename Iterator1, typename Iterator2, typename Iterator3, typename Size,
+              typename BinaryPredicateCheck = oneapi::dpl::__internal::__pstl_equal,
               typename BinaryOperationCheck = oneapi::dpl::__internal::__pstl_plus>
-    void
-    check_values(Iterator1 host_keys, Iterator2 host_vals, Iterator3 val_res, Size n,
-                 BinaryOperationCheck op = BinaryOperationCheck())
+    void check_values(Iterator1 host_keys, Iterator2 host_vals, Iterator3 val_res, Size n,
+                      BinaryPredicateCheck pred = BinaryPredicateCheck(),
+                      BinaryOperationCheck op = BinaryOperationCheck())
     {
         // https://docs.oneapi.io/versions/latest/onedpl/extension_api.html
         // keys:   [ 0, 0, 0, 1, 1, 1 ]
         // values: [ 1, 2, 3, 4, 5, 6 ]
         // result: [ 1, 1 + 2 = 3, 1 + 2 + 3 = 6, 4, 4 + 5 = 9, 4 + 5 + 6 = 15 ]
 
-#ifdef DUMP_CHECK_RESULTS
-        std::cout << "check_values(n = " << n << ") : " << std::endl;
-        display_param("keys:   ", host_keys, n);
-        display_param("values: ", host_vals, n);
-        display_param("result: ", val_res,   n);
-#endif // DUMP_CHECK_RESULTS
-
         if (n < 1)
             return;
 
-        using ValT = typename ::std::decay<decltype(val_res[0])>::type;
+        typedef typename ::std::iterator_traits<Iterator1>::value_type KeyT;
+        typedef typename ::std::iterator_traits<Iterator2>::value_type ValT;
+
+        ::std::ostringstream failure_msg;
+        failure_msg << "wrong effect from inclusive_scan_by_segment with types [";
+        failure_msg << typeid(KeyT).name();
```

Comment:
Do we relay need a kind of RTTI here? (`typeid`)

Reply:
I will remove this. I do not think it is necessary for the tests now since the branch is stable.